File: chat/src/main/kotlin/chatox/chat/api/request/UpdateChatNotificationsSettingsRequest.kt
package chatox.chat.api.request

import chatox.chat.model.NotificationLevel
import chatox.chat.model.NotificationSound
import com.fasterxml.jackson.annotation.JsonProperty
import jakarta.validation.constraints.NotNull

data class UpdateChatNotificationsSettingsRequest(
        @field:JsonProperty("level")
        @field:NotNull
        private val _level: NotificationLevel,

        val sound: NotificationSound,
        val userExceptions: Map<String, UpdateNotificationsSettingsRequest>? = null
) {
    val level: NotificationLevel
        get() = _level!!
}

File: client/src/api/types/request/UpdateChatNotificationsSettingsRequest.ts
import {UpdateNotificationsSettingsRequest} from "./UpdateNotificationsSettingsRequest";
import {NotificationLevel, NotificationSound} from "../response";

export interface UpdateChatNotificationsSettingsRequest {
	level: NotificationLevel,
	sound: NotificationSound,
	userExceptions?: {
		[userId: string]: UpdateNotificationsSettingsRequest
	}
}
